integration-tests/pkg/common/utils.go

```diff
@@ -64,3 +67,9 @@ func PrepareLog(testName string, logName string) (*os.File, error) {
 	logPath := filepath.Join(logDirectory, strings.ReplaceAll(testName, "/", "_")+"-"+logName)
 	return os.Create(logPath)
 }
+
+func Sleep(duration time.Duration) {
```

Comment:
Another change that's not strictly related to the bug, but is helpful to understand the sleep behaviour of the tests (may help us to narrow down timing flakes in the future)

Reply:
Like this one, it should help reducing need in any kind of waiting in tests.